implementations/java-jsonschema-validator/bowtieJsonSchemaValidator.java

```diff
@@ -0,0 +1,215 @@
+import java.util.jar.Manifest;
+
+public class BowtieJsonSchemaValidator {
+
+  private static final List<String> DIALECTS =
+      List.of("https://json-schema.org/draft/2020-12/schema",
+              "https://json-schema.org/draft/2019-09/schema",
+              "http://json-schema.org/draft-07/schema#",
+              "http://json-schema.org/draft-06/schema#",
+              "http://json-schema.org/draft-04/schema#");
+
+  private final ObjectMapper objectMapper = new ObjectMapper().configure(
+      DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+  private final JsonSchemaFactory schemaFactory =
+      JsonSchemaFactory.getInstance();
+  private final PrintStream output;
+
+  public static void main(String[] args) {
+    BufferedReader reader =
+        new BufferedReader(new InputStreamReader(System.in));
+    new BowtieJsonSchemaValidator(System.out).run(reader);
+  }
+
+  public BowtieJsonSchemaValidator(PrintStream output) { this.output = output; }
+
+  private void run(BufferedReader reader) {
+    reader.lines().forEach(this::handle);
+  }
+
+  private void handle(String data) {
+    try {
+      JsonNode node = objectMapper.readTree(data);
+      String cmd = node.get("cmd").asText();
+      switch (cmd) {
+        case "start" -> start(node);
+        case "dialect" -> dialect(node);
+        case "run" -> run(node);
+        case "stop" -> System.exit(0);
+        default -> throw new IllegalArgumentException(
+          "Unknown cmd [%s]".formatted(cmd)
+        );
+      }
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private void start(JsonNode node) throws IOException {
+    StartRequest startRequest = objectMapper.treeToValue(
+      node,
+      StartRequest.class
+    );
+    if (startRequest.version() != 1) {
+      throw new IllegalArgumentException(
+        "Unsupported IHOP version [%d]".formatted(startRequest.version())
+      );
+    }
+
+    InputStream is = getClass().getResourceAsStream("META-INF/MANIFEST.MF");
+    var attributes = new Manifest(is).getMainAttributes();
+
+    StartResponse startResponse = new StartResponse(
+      1,
+      true,
+      new Implementation(
+        "java",
+        attributes.getValue("Implementation-Name"),
+        attributes.getValue("Implementation-Version"),
+        DIALECTS,
+        "https://github.com/networknt/json-schema-validator/",
+        "https://github.com/networknt/json-schema-validator/issues",
+        System.getProperty("os.name"),
+        System.getProperty("os.version"),
+        System.getProperty("java.vendor.version")
+      )
+    );
+    output.println(objectMapper.writeValueAsString(startResponse));
+  }
+
+  private void dialect(JsonNode node) throws JsonProcessingException {
+    DialectRequest dialectRequest = objectMapper.treeToValue(
+      node,
+      DialectRequest.class
+    );
+    if (DIALECTS.contains(dialectRequest.dialect())) {
+      output.println(
```

Comment:
When setting specific dialect you need to perform some action on the validator. It looks like you should call e.g. `JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V4)`, but ofc with specific dialect

Reply:
I think we should maintain the `dialect` method exclusively for configuring the dialects without the `$schema` property. All actions associated with validation should be added in the `run` method.